objectivec/test/ort_training_session_test.mm

```diff
@@ -0,0 +1,384 @@
+                                                          evalModelPath:[ORTTrainingSessionTest getFilePathFromName:@"eval_model.onnx"]
+                                                     optimizerModelPath:[ORTTrainingSessionTest getFilePathFromName:@"adamw.onnx"]
+                                                                  error:&error];
+
+  ORTAssertNullableResultSuccessful(session, error);
+  return session;
+}
+
+- (void)testInitTrainingSession {
+  NSError* error = nil;
+
+  ORTCheckpoint* checkpoint = [[ORTCheckpoint alloc] initWithPath:[ORTTrainingSessionTest getFilePathFromName:@"checkpoint.ckpt"]
+                                                            error:&error];
+  ORTAssertNullableResultSuccessful(checkpoint, error);
+  ORTTrainingSession* session = [self makeTrainingSessionWithCheckPoint:checkpoint];
+
+  // check that inputNames contains input-0
+  NSArray<NSString*>* inputNames = [session inputNamesWithTraining:YES error:&error];
+  ORTAssertNullableResultSuccessful(inputNames, error);
+
+  XCTAssertTrue(inputNames.count > 0);
+  XCTAssertTrue([inputNames containsObject:@"input-0"]);
+
+  // check that outNames contains onnx::loss::21273
+  NSArray<NSString*>* outputNames = [session outputNamesWithTraining:YES error:&error];
+  ORTAssertNullableResultSuccessful(outputNames, error);
+
+  XCTAssertTrue(outputNames.count > 0);
+  XCTAssertTrue([outputNames containsObject:@"onnx::loss::21273"]);
+}
+
+- (void)testInintTrainingSessionWithEval {
+  NSError* error = nil;
+  ORTCheckpoint* checkpoint = [[ORTCheckpoint alloc] initWithPath:[ORTTrainingSessionTest getFilePathFromName:@"checkpoint.ckpt"] error:&error];
+  ORTAssertNullableResultSuccessful(checkpoint, error);
+  ORTTrainingSession* session = [self makeTrainingSessionWithCheckPoint:checkpoint];
+
+  // check that inputNames contains input-0
+  NSArray<NSString*>* inputNames = [session inputNamesWithTraining:NO error:&error];
+  ORTAssertNullableResultSuccessful(inputNames, error);
+
+  XCTAssertTrue(inputNames.count > 0);
+  XCTAssertTrue([inputNames containsObject:@"input-0"]);
+
+  // check that outNames contains onnx::loss::21273
+  NSArray<NSString*>* outputNames = [session outputNamesWithTraining:NO error:&error];
+  ORTAssertNullableResultSuccessful(outputNames, error);
+
+  XCTAssertTrue(outputNames.count > 0);
+  XCTAssertTrue([outputNames containsObject:@"onnx::loss::21273"]);
+}
+
+- (void)runTrainStepWithSession:(ORTTrainingSession*)session {
+  // load input and expected output
+  NSError* error = nil;
+  NSMutableData* expectedOutput = [ORTTrainingSessionTest loadTensorFromFile:[ORTTrainingSessionTest getFilePathFromName:@"loss_1.out"] skipHeader:YES];
+
+  NSMutableData* input = [ORTTrainingSessionTest loadTensorFromFile:[ORTTrainingSessionTest getFilePathFromName:@"input-0.in"] skipHeader:YES];
+
+  int32_t lables[] = {1, 1};
+
+  // create ORTValue array for input and labels
+  NSMutableArray<ORTValue*>* pinnedInputs = [NSMutableArray array];
```

Comment:
Thank you! I think `pinned` was used because C# and Java API requires names for the Input. Since, Objective-C API does not need the names, I think name like `inputValues` will make more sense.